### DISCUSSION TITLE
RabbitMQ Service stops after upgrading from 3.10.8 to 3.12.3

### DISCUSSION THREAD

**Discussion by deepankarrelativity002** [rabbitmq/rabbitmq-server#9690] [Questions] [answered]
In one of my instances, after upgrading RabbitMQ to 3.12.3, service(RabbitMQ) stops after 5-10 sec, hence throwing an error while connecting to API. Also, usually RabbitMQ folder gets generated other than installation folder where we can check but for this case it's not getting generated. How to fix this? Compatible Erlang(26.0.2) is being used

**Comment by mkuratczyk** [comment]
You provided no information whatsoever that would allow anyone to help you. What error? What's in the logs? Did you upgrade directly from 3.10 to 3.12 (that's not supported)? What do you mean by "folder gets generated"?

Did you follow the upgrade guide, including enabling feature flags?

**Reply by deepankarrelativity002** [reply]
Log file is not getting generated after upgrading it to 3.12.x to check out for the errors

**Comment by michaelklishin** [comment]
If the log file is not created it may mean that the node terminates very early in its boot process. There should be at least some output in relevant service/OS logs.

Also, are you sure that your node has all the feature flags enabled? This is an upgrade requirement for 3.12, as mentioned in release notes. If that's not the case, nodes will fail to start.

**Reply by michaelklishin** [reply, marked as answer]
@senimai if this hypothesis is correct, you have two options:

1. Go back to 3.11.23, [enable all feature flags](https://rabbitmq.com/feature-flags.html) and then upgrade to 3.12 again
2. If this is a development installation, stopping the node, wiping [node's data directory](https://rabbitmq.com/relocate.html) should make the node happily start again. Then you can enable all feature flags to be more future-proof

**Reply by dumbbell** [reply]
If the previous version was 3.10.8, you will have to downgrade to 3.10.8, enable all feature flags, then upgrade to 3.11.23 and do the same, before upgrading one last time to 3.12.3.

**Comment by deepankarrelativity002** [comment]
Followed the steps being given by @michaelklishin

**Reply by michaelklishin** [reply]
Did they help? :)
